src/main/java/de/tum/in/www1/artemis/web/rest/ModelingAssessmentResource.java

```diff
@@ -286,4 +297,19 @@ public ResponseEntity cancelAssessment(@PathVariable Long submissionId) {
     String getEntityName() {
         return ENTITY_NAME;
     }
+
+    /**
+     * Checks if the current user is allowed to retrieve the given result. The user is allowed to retrieve the result if he is a student in the corresponding course, the submission
+     * is his submission, the assessment due date of the corresponding exercise is in the past (or not set) and the result is finished.
+     *
+     * @param exercise      the corresponding exercise
+     * @param participation the participation the result belongs to
+     * @param result        the result that should be sent to the client
+     * @return true if the user is allowed to retrieve the given result, false otherwise
+     */
+    private boolean isUserAllowedToGetAssessment(Exercise exercise, Participation participation, Result result) {
```

Comment:
Would it be possible to move this to the AuthorizationCheckService?
Also there is a similiar method to `courseService.userHasAtLeastStudentPermissions(exercise.getCourse())` in AuthorizationCheckService.isStudentInCourse.

I sometimes wonder if we should try to centralize the permissions more into one place to enforce reuse  :thinking:.

Reply:
You are right, I moved the method to the `AuthorizationCheckService`.